activestorage/app/models/active_storage/preview.rb

```diff
@@ -53,7 +53,11 @@ def processed
 
   # Returns the blob's attached preview image.
   def image
-    blob.preview_image
+    if variant?
+      variant.processed.image if processed?
+    else
+      preview_image
+    end
```

Comment:
Although this is arguably the desired behavior, this would be a breaking change.  Since this method is part of the public API, we would need to implement some sort of deprecation if we went this route.

Reply:
I just got and idea to avoid this deprecation and fix `AS::Representations::ProxyController`.

since `Variant` is already defining the missing methods to make itself a `ActiveStorage::Blob` compliant ruby model. 

how about we change : 
```ruby
send_blob_stream @representation.image, disposition: params[:disposition]
```
to 
```ruby 
send_blob_stream @representation, disposition: params[:disposition]
```

Then what's left is to add to both `VariantWithRecord` and `Preview` these : 

```ruby
delegate :forced_disposition_for_serving, :forced_disposition_for_serving, :filename, to: :image
```

This way:
1- We make all three classes truly compatible and implementing all the methods
2- Fix `ProxyController` for all the 3 representables.
3- We can separate fixing `Preview#image` in another refactoring PR if needed

@jonathanhefner